src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -977,6 +990,50 @@ export function calculateLayout(visibleAttributes: ChartMeta[], cols: number): L
     return layout;
 }
 
+export function populateFixedPositionLayout(layout: Layout[], matrix: string[][], currentGridLayout: Layout, cols: number, currentGridElement: ChartMeta){
```

Comment:
Since this is an util function, it should be separated to few steps otherwise you are losing the context.
- Populate matrix by using currentGridElement (This probably can be an util function)
- Add the currentGridElement to layout array(this can just be added in calculateLayout method)

Reply:
All comment has been applied to the updated code in the second commit. Thanks!